standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/messaging/event/filters/CatalogFilter.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.metastore.messaging.event.filters;
+
+import org.apache.hadoop.hive.metastore.api.NotificationEvent;
+
+/**
+ * Utility function that constructs a notification filter to match a given catalog name.
+ */
+public class CatalogFilter extends BasicFilter {
+  private final String catalogName;
+
+  public CatalogFilter(final String catalogName) {
+    this.catalogName = catalogName;
+  }
+
+  @Override
+  boolean shouldAccept(final NotificationEvent event) {
+    if (catalogName == null || event.getCatName() == null || catalogName.equalsIgnoreCase(event.getCatName())) {
```

Comment:
catalogName should never be null. If not configured also, it must be default which is "hive". I think we should let it fail if filter is initialized with catalogName being null .